packages/frontend/src/creation-form/index.tsx

```diff
@@ -184,32 +184,31 @@ export const Component = ({
             !isNaN(parseInt(minimumBond)) &&
             openingTimestamp.isAfter(dayjs())
         ) {
-            const formattedMinimumBond = ethers.utils.parseUnits(
-                minimumBond,
+            const formattedMinimumBond = parseUnits(
+                minimumBond as `${number}`,
                 chain.nativeCurrency.decimals
             );
             initializationDataGetter = async () => {
                 const questionCid = await uploadToIpfs(question);
-                const initializationData = ethers.utils.defaultAbiCoder.encode(
+                const initializationData = encodeAbiParameters(
                     [
-                        "address",
-                        "uint256",
-                        "string",
-                        "uint32",
-                        "uint32",
-                        "uint256",
+                        { type: "address", name: "arbitratorAddress" },
+                        { type: "uint256", name: "realityTemplateId" },
+                        { type: "string", name: "questionIdentifier" },
+                        { type: "uint32", name: "questionTimeout" },
+                        { type: "uint32", name: "openingTimestamp" },
+                        { type: "uint256", name: "minimumBond" },
                     ],
                     [
-                        arbitrator.value,
-                        realityTemplateId.value,
+                        arbitrator.value as `0x${string}`,
+                        BigInt(realityTemplateId.value),
                         `${questionCid}-${realityTemplateId.value}`,
-                        questionTimeout.value,
+                        Number(questionTimeout.value),
```

Comment:
Ditto here

Reply:
Same reasons a the previous comment here.